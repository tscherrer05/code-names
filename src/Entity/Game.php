<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\GameRepository")
 */
class Game
{
    /**
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="guid")
     */
    private $publicKey;

    /**
     * @ORM\Column(type="integer")
     */
    private $status;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $currentWord;

    /**
     * @ORM\Column(type="integer", nullable=true)
     */
    private $currentNumber;

    /**
     * @ORM\Column(type="smallint", nullable=true)
     */
    private $currentTeam;

    /**
     * @ORM\OneToMany(targetEntity="App\Entity\Card", mappedBy="game", fetch="EAGER")
     */
    private $cards;

    /**
     * @ORM\OneToMany(targetEntity="App\Entity\GamePlayer", mappedBy="game", fetch="EAGER")
     */
    private $gamePlayers;

    public function __construct()
    {
        $this->cards = new ArrayCollection();
        $this->gamePlayers = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPublicKey(): ?string
    {
        return $this->publicKey;
    }

    public function setPublicKey(string $publicKey): self
    {
        $this->publicKey = $publicKey;

        return $this;
    }

    public function getStatus(): ?int
    {
        return $this->status;
    }

    public function setStatus(int $status): self
    {
        $this->status = $status;

        return $this;
    }

    public function getCurrentWord(): ?string
    {
        return $this->currentWord;
    }

    public function setCurrentWord(?string $currentWord): self
    {
        $this->currentWord = $currentWord;

        return $this;
    }

    public function getCurrentNumber(): ?int
    {
        return $this->currentNumber;
    }

    public function setCurrentNumber(?int $currentNumber): self
    {
        $this->currentNumber = $currentNumber;

        return $this;
    }

    public function getCurrentTeam(): ?int
    {
        return $this->currentTeam;
    }

    public function setCurrentTeam(?int $currentTeam): self
    {
        $this->currentTeam = $currentTeam;

        return $this;
    }

    /**
     * @return Collection|Card[]
     */
    public function getCards(): Collection
    {
        return $this->cards;
    }

    public function addCard(Card $card): self
    {
        if (!$this->cards->contains($card)) {
            $this->cards[] = $card;
            $card->setGame($this);
        }

        return $this;
    }

    public function removeCard(Card $card): self
    {
        if ($this->cards->contains($card)) {
            $this->cards->removeElement($card);
            // set the owning side to null (unless already changed)
            if ($card->getGame() === $this) {
                $card->setGame(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection|GamePlayer[]
     */
    public function getGamePlayers(): Collection
    {
        return $this->gamePlayers;
    }

    public function addGamePlayer(GamePlayer $gamePlayer): self
    {
        if (!$this->gamePlayers->contains($gamePlayer)) {
            $this->gamePlayers[] = $gamePlayer;
            $gamePlayer->setGame($this);
        }

        return $this;
    }

    public function removeGamePlayer(GamePlayer $gamePlayer): self
    {
        if ($this->gamePlayers->contains($gamePlayer)) {
            $this->gamePlayers->removeElement($gamePlayer);
            // set the owning side to null (unless already changed)
            if ($gamePlayer->getGame() === $this) {
                $gamePlayer->setGame(null);
            }
        }

        return $this;
    }
}
