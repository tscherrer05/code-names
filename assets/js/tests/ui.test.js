import React from 'react'
import renderer from 'react-test-renderer'
import Game from '../game'

jest.mock('../DataSource')

it('game UI renders correctly', () => {
    const gameKey = '123'
    const playerKey = '456'
    const tree = renderer
        .create(<Game gameKey={gameKey} playerKey={playerKey}></Game>)
        .toJSON()
    expect(tree).toMatchSnapshot()
});