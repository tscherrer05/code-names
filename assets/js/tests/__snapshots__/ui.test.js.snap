// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game UI renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      Menu
    </a>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        <div
          className="row"
          id="cn-cards-row"
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <p>
            Vous êtes : 
            <span
              id="current-player"
            />
          </p>
          <p>
            Votre équipe : 
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              Rouge
            </span>
          </p>
          <p>
            Votre rôle : 
            Maître espion
          </p>
        </div>
        <div
          className="col"
        >
          <p>
            Tour : 
            <span
              id="current-team"
            >
              Equipe 
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                Rouge
              </span>
            </span>
          </p>
          <span
            data-value="123"
            id="gameKey"
          />
          <span
            data-value="456"
            id="playerKey"
          />
        </div>
        <div
          className="col"
        >
          <p
            className="display-block"
          >
            Votes restants :
          </p>
        </div>
        <div
          className="col"
        >
          <p
            className="display-block"
          >
            Joueurs connectés :
          </p>
          <ul />
        </div>
      </div>
    </div>
  </div>
</div>
`;
